### ISSUE TITLE
The search result may be outdated. Run `git pull` to get the latest results.

### ISSUE THREAD

**Issue by keyvan136181** [microsoft/vcpkg#19237]
Install PETSc library with vcpkg.

**Comment by autoantwort** [comment]
What is your question?

**Comment by JonLiu1993** [comment]
@keyvan136181 ,Thanks for posting this issue.
Could you describe your issue in detail, as well as the reproduction environment and reproduction steps? And provide a complete error log

**Comment by szabolcshorvath-craft** [comment]
I have the same problem. OS: Windows 10 x64. Steps to reproduce and further environment details below. Where can one find the complete error logs? I'd be most happy to provide one, but you didn't say which file you wanted to see. (It looks like a warning or note but not an error, by the way.)

<pre>
C:\Users\szabolcs\vcpkg>git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

C:\Users\szabolcs\vcpkg>vcpkg --version
Vcpkg package management program version 2021-07-26-9425cf5f512f242c0bcbabac31f08832825aee81

See LICENSE.txt for license information.

C:\Users\szabolcs\vcpkg>vcpkg search gdk
gdk-pixbuf               2.42.2#1         Image loading library.
<b><u>The search result may be outdated. Run `git pull` to get the latest results.</u></b>

If your library is not listed, please open an issue at and/or consider making a pull request:
    https://github.com/Microsoft/vcpkg/issues

C:\Users\szabolcs\vcpkg>git pull
Already up to date.

C:\Users\szabolcs\vcpkg>vcpkg search gdk
gdk-pixbuf               2.42.2#1         Image loading library.
<b></u>The search result may be outdated. Run `git pull` to get the latest results.</u></b>

If your library is not listed, please open an issue at and/or consider making a pull request:
    https://github.com/Microsoft/vcpkg/issues

C:\Users\szabolcs\vcpkg>
<pre>

**Comment by autoantwort** [comment]
Yeah it is a warning, but we don't check anything. If we don't want to display the warning we have to do a git fetch for every search but this is way to expensive.

**Comment by szabolcshorvath-craft** [comment]
In that case (prohibite cost of checks), it would be feasible to check that *only once a week* or another period that matches the average frequency of changes in the repository. (Additionally, an ls-remote operation might be more cheap than a fetch but I'm not an expert in git TBH.)
Another alternative would be to give users the choice of opting out of seeing that message, as it would not be unreasonable to expect them to be aware of this aspect of package managers:.`vcpkg --im-aware-that-package-descriptions-are-always-getting-outdated-lol ...` and the message would not appear anymore.

**Comment by autoantwort** [comment]
> or another period that matches the average frequency of changes in the repository

Yeah changes happens every few hours. So if the last git pull is one hour old, the result is maybe outdated again. But yeah once a week is properly enough. The problem was that people often requested new versions of libs that were already there but they have not updated their vcpkg instance. 

ls-remote needed 2 seconds for me. Not optimal.  

> vcpkg --im-aware-that-package-descriptions-are-always-getting-outdated-lol

Can you then not simply ignore the message. 

But maybe one could use ` git --no-pager log -1 --format=%cd ` to then display a message like: `The search results are x days old. Use git pull to get the latest results`

**Comment by JackBoosY** [comment]
That's just a notice.

**Comment by dg0yt** [comment]
IMO recommending a "git pull" is too intrusive. I would prefer to have a `--online` option. which could be used (and recommeded) safely.

**Comment by autoantwort** [comment]
The --online option should then simply open the webbrowser and search on vcpkg.io?

> get the latest (???) results.

If they have not changed it no. They have to update the site manually... WTF

**Comment by JackBoosY** [comment]
@autoantwort  Yes, it is always up to date only in our master branch.

**Comment by autoantwort** [comment]
> @autoantwort Yes, it is always up to date only in our master branch.

you mean in the vcpkg repo? The vcpkg.io repo is definitely not up to date. For example vcpkg.io still shows the old version for FreeRDP which was updated in #18242

**Comment by JackBoosY** [comment]
@autoantwort Yes, in vcpkg.io, we are not updated synchronously.

**Comment by JonLiu1993** [comment]
@keyvan136181 ,That's just a notice.Can this reply solve your problem？
